examples/topics/categorical/correlograms.py

```diff
@@ -0,0 +1,68 @@
+''' A categorical plot showing the correlations in mineral content for 214 samples of glass fragments
+obtained during forensic work. The dataset contains seven variables measuring the amounts of magnesium (Mg),
+calcium (Ca), iron (Fe), potassium (K), sodium (Na), aluminum (Al), and barium (Ba) found in each glass fragment.
+The colored circles represent the correlations between pairs of these variables.
+The magnitude of each correlation is encoded in the size of the circles.
+
+.. bokeh-example-metadata::
+    :sampledata: forensic_glass
+    :apis: bokeh.plotting.figure.scatter
+    :refs: :ref:`ug_topics_categorical_scatters`
+    :keywords: scatter, correlogram
+
+'''
+import numpy as np
+import pandas as pd
+
+from bokeh.models import ColorBar, ColumnDataSource, FixedTicker, LinearColorMapper
+from bokeh.plotting import figure, show
+from bokeh.sampledata.forensic_glass import data
+from bokeh.transform import transform
+
+df = data.copy()
+
+pairs = ["Mg-Ba", "Mg-Al", "Mg-Na", "Mg-K", "Mg-Fe", "Mg-Ca", "Ca-Ba",
+         "Ca-Al", "Ca-Na", "Ca-K", "Ca-Fe", "Fe-Ba", "Fe-Al", "Fe-Na",
+         "Fe-K", "K-Ba", "K-Al", "K-Na", "Na-Ba", "Na-Al", "Al-Ba"]
+
+pair_split = [pair.split("-") for pair in pairs]
+correlations = []
+for pair in pair_split:
+    matrix = np.corrcoef(df[f"{pair[0]}"], df[f"{pair[1]}"])
+    correlation = matrix[0, 1]
+    correlations.append(correlation)
+
+new_df = pd.DataFrame({"oxide_1": [x[0] for x in pair_split], "oxide_2": [x[1] for x in pair_split],
+                       "correlation": correlations, "dot_size": [abs(corr) * 120 for corr in correlations]})
+
+x_range = new_df["oxide_1"].unique()
+y_range = list(reversed(new_df["oxide_2"].unique()))
+
+source = ColumnDataSource(new_df)
+
+colors = ["#8B4513", "#D8AF85", "#CD853F", "#E2CAB7", "#DEB887",
+          "#A4D2D2", "#8ABDBD", "#ADD8E6", "#009999", "#188A8A"]
+
```

Comment:
For the purposes of the blog series, we were trying to reproduce things very exactly but for the purpose of a new example in the repo, I think it would be better to pick an existing colormap and use that.

Reply:
I think we'll need to experiment. The very dark ends of the new palette are actually very hard to distinguish for me. I think that palette might do ok in contour map where things vary "continously" but for isolated circles floating in a light background, they just look like "dark circles" on either side. Maybe `RdYlGn10` or `Sunset10` or even `Iridescent10` (on a darker background)? If we go down to 9 bins instead of 10 that would also open up more possibilities eg. `TolPRGn` or `BuRd`

Reply:
An other option might be `Turbo` which is an improved perceptually uniform "rainbow-like"

https://docs.bokeh.org/en/latest/docs/reference/palettes.html#large-palettes

I think the ends might still be extreme, isolated  on a light background, we could consider using just the middle part, e.g. `Turbo256[50:-50]` or something. 

Play around with a few of these options and see if you can find a couple to propose.

Reply:
It seems `Turbo256` is a darker version of `Sunset`, but here are a few colors I tried...

`Iridescent` on a dark background:
<img width="613" alt="Screenshot 2023-10-04 at 10 17 13 PM" src="https://github.com/bokeh/bokeh/assets/99359668/0d656f2b-f1f6-40de-aa2e-2b1602e6a600">

`RdYlGn`
<img width="599" alt="Screenshot 2023-10-04 at 10 18 46 PM" src="https://github.com/bokeh/bokeh/assets/99359668/b4585a4e-4cd7-436a-9cd8-e26d6704d7b1">

`Turbo`
<img width="589" alt="Screenshot 2023-10-04 at 10 21 34 PM" src="https://github.com/bokeh/bokeh/assets/99359668/fcbd5810-aacd-447a-b2d3-cc22bf727f3e">

`Sunset`
<img width="593" alt="Screenshot 2023-10-04 at 10 22 58 PM" src="https://github.com/bokeh/bokeh/assets/99359668/5eab5905-33d8-4f2b-88ff-f8b786260752">

Reply:
Of those, thoughts:

* I like RdYlGn alot, but the small light circles are a little faint. How would thing look if we add a thin black border to the circles?
*  Sunset looks OK to me
* Turbo also looks OK but I think it would be better if we lopped off some of the palette at either end